### ISSUE TITLE
Accuracy on AlexNet validation set with conv4 feataure

### ISSUE THREAD

**Issue by cao-nv** [gidariss/FeatureLearningRotNet#10]
Hi, 
I just re-implement your idea myself by following the details in this repository. 
The experiments on CIFAR10 obtained about 1% lower accuracy than published results. 
However, the experiments on ImageNet with AlexNet achieved about 3% higher accuracy on the ImageNet validation set than the published results. 

`supervised`: 59.48 (59.70 in the paper)

`conv4`: 52.92 (50 in the paper) 

`conv5`: 46.06 (43.8 in the paper) 

Could you give more details about the training?

**Comment by gidariss** [comment]
Hello, you could see the training details on the code. The main thing are the hyper-pameters, for the imagenet model see here:
https://github.com/gidariss/FeatureLearningRotNet/blob/master/config/ImageNet_RotNet_AlexNet.py
and the dataloader:
https://github.com/gidariss/FeatureLearningRotNet/blob/master/dataloader.py

**Comment by cao-nv** [comment]
> 
> 
> Hello, you could see the training details on the code. The main thing are the hyper-pameters, for the imagenet model see here:
> https://github.com/gidariss/FeatureLearningRotNet/blob/master/config/ImageNet_RotNet_AlexNet.py
> and the dataloader:
> https://github.com/gidariss/FeatureLearningRotNet/blob/master/dataloader.py

Thank you for your reply, 

I have spotted that the results reported in the paper is from a 30-epoch model, but in your implementation (also my re-implementation) the number of epochs is set to 50. That may be the reason for the difference. 

Actually, I have not read all the papers in this unsupervised representation learning task. So I'm curious about the number of epochs you used. Does it have to be 30 to get fair comparisons with the previous works?

**Comment by gidariss** [comment]
No, it has not to be 30 epochs. How many epochs you will use is entirely up you. You will probably choose the number of epoch that are required for convergence. I initially used 30 epochs in order to have the results ready for the paper submission.

**Comment by cao-nv** [comment]
Thank you. I think you did want to train the non-linear classifier more because your method already beat the state-of-the-art methods at epoch 30.
